### ISSUE TITLE
[VRL] Inconsistent error reporting when using path notation in function parameters

### ISSUE THREAD

**Issue by hhromic** [vectordotdev/vector#7717]
### Vector Version

```
vector 0.13.1 (v0.13.1 x86_64-unknown-linux-gnu 2021-04-29)
```

### Description

Consider the following sequence of instructions in `vector vrl`:
```
$ nothing
null

$ nothing.nothing
null

$ split!(nothing, "=")

error[E110]: invalid argument type
  ┌─ :1:8
  │
1 │ split!(nothing, "=")
  │        ^^^^^^^
  │        │
  │        this expression resolves to the exact type "null"
  │        but the parameter "value" expects the exact type "string"
  │

$ split!(nothing.nothing, "=")
function call error for "split" at (0:28): expected "null", got "string"

$ parse_key_value!(nothing.nothing)
function call error for "parse_key_value" at (0:33): expected "null", got "string"

$ something = {}
{  }

$ parse_key_value!(something.nothing)
function call error for "parse_key_value" at (0:35): expected "null", got "string"
```

On the one hand, I would expect all these example function calls to error out as the first example.
But on the other hand, the second error is very cryptic and misleading.

We found this error while processing some data with unexpected missing fields, and Vector was reporting the latter errors in the example above. We got very confused by the *expected null, got string* message until we realised it was a non-existing field on a non-null object (the very last example above).

**Comment by jszwedko** [comment]
Thanks for the report @hhromic . This bug was actually fixed in https://github.com/timberio/vector/pull/6985/files#diff-e563f1f5638011801a2bea62a515a9628e11a68dd9afbe2a72f898aa992a8e03 and released in 0.14.0.

It seems like the error messages are correct now:

```
$ split!(nothing.nothing, "=")
function call error for "split" at (0:28): expected "string", got "null"

$ something = {}
{  }

$ parse_key_value!(something.nothing)
function call error for "parse_key_value" at (0:35): expected "string", got "null"
```

I'll close this out, but please reopen if you still see issues.

**Comment by hhromic** [comment]
@jszwedko Yep! can confirm the error message is correct now in 0.14.0. Thanks!
I normally monitor PRs in Vector just to keep informed of what is coming/being fixed but I missed this one :)
